pkg/log/log.go

```diff
@@ -108,3 +112,38 @@ func parseLogLevel(logLevel string) (level.Option, error) {
 		return nil, fmt.Errorf("unrecognized log level %q", logLevel)
 	}
 }
+
+const logOnceTimedDuration = time.Second * 10
+
+// timedLogger logs from logStore every logOnceTimedDuration. It deletes the log entry from the store
+// after it has logged once.
+func timedLogger() {
+	t := time.NewTicker(logOnceTimedDuration)
+	for range t.C {
+		logStoreMux.Lock()
+		for k, logMsg := range logStore {
+			Debug("msg", "timed-debug")
+			Debug(logMsg...)
+			delete(logStore, k)
+		}
+		if len(logStore) == 0 {
```

Comment:
This will always, be true no?

Reply:
in fact I'm ok with a panic if this isn't true

Reply:
Yeah, this should always be true. Here, just for sanitary checks.